recipes/vectorclass/all/conanfile.py

```diff
@@ -50,12 +73,35 @@ def lazy_lt_semver(v1, v2):
             raise ConanInvalidConfiguration("{} {} requires C++17, which your compiler does not support.".format(self.name, self.version))
 
     def package_id(self):
-        self.info.header_only()
+        if self.options.header_only:
+            self.info.header_only()
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version],
                   destination=self._source_subfolder, strip_root=True)
 
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        if not self.options.header_only:
+            cmake = self._configure_cmake()
+            cmake.build()
+
     def package(self):
         self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy("*.h", dst="include", src=self._source_subfolder)
+        if self.options.header_only:
+            self.copy("*.h", dst="include", src=self._source_subfolder)
+        else:
+            cmake = self._configure_cmake()
+            cmake.install()
+
+    def package_info(self):
+        if self.options.header_only:
+            self.cpp_info.defines = ["VECTORCLASS_HEADER_ONLY"]
```

Comment:
Honestly I don't like this custom definition in package_info(), consumers shouldn't rely on it.